### PR TITLE
Feature/css proceeding

### PR DESCRIPTION
**Pull request by b1ank04**

### CODE REVIEW

File: src/main/java/de/ace/html2pdf/controller/PdfGenerationController.java [hunk 1]
```diff
@@ -22,26 +23,37 @@
...
+    private final PdfRenderComponent pdfRenderComponent;
 
     @GetMapping("/healthCheck")
     public ResponseEntity<String> healthCheck() {
         return ResponseEntity.ok("\uD83D\uDD25");
     }
 
     @PostMapping("/html")
-    public ResponseEntity<byte[]> generatePdfHtml(@RequestBody PdfRequest pdfRequest) {
+    public ResponseEntity<byte[]> generatePdfHtml(@RequestBody String html) {
```
[the remark this answers is not in the document]
Comment (b1ank04) [reply src/main/java/de/ace/html2pdf/controller/PdfGenerationController.java:35]: It is a technical requirement

File: src/main/java/de/ace/html2pdf/model/PdfData.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package de.ace.html2pdf.model;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class PdfData {
```
[the remark this answers is not in the document]
Comment (b1ank04) [reply src/main/java/de/ace/html2pdf/model/PdfData.java:8]: It has a @Builder pattern already, do you mean something else?

File: src/main/java/de/ace/html2pdf/MagicApplication.java [hunk 3]
```diff
@@ -1,14 +1,13 @@
 package de.ace.html2pdf;
 
-import org.jsoup.Jsoup;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
 public class MagicApplication {
```
[the remark this answers is not in the document]
Comment (KirkDerAuslaender) [reply src/main/java/de/ace/html2pdf/MagicApplication.java:7]: I like MagicApplication.
Actually we would rename it to: UltraMagicalApplication

File: src/main/java/de/ace/html2pdf/model/FooterProperties.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package de.ace.html2pdf.model;
+
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+@RequiredArgsConstructor
+public class FooterProperties {
```
[the remark this answers is not in the document]
Comment (KirkDerAuslaender) [reply src/main/java/de/ace/html2pdf/model/FooterProperties.java:8]: Actually, we could use a record with @With.
Therefore it is immutable but you are able to compose a new record with FooterImageProperties.

i.e: @With
record Whatever { }

File: src/main/java/de/ace/html2pdf/model/PdfData.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (KirkDerAuslaender) [reply src/main/java/de/ace/html2pdf/model/PdfData.java:8]: Again on this one. Wither is for inmmutable objects (Objects that will go across your application, consistency principle). Builder is for objects that will need to be used shortly and consumed and the 1st or 2nd forward layer. However, I would also recommend to have a consistency across the same model structure. Therefore, if we want to have both approaches and we have lots of model, I would recommend to package them within the model by domain

File: src/main/java/de/ace/html2pdf/controller/PdfGenerationController.java [hunk 5]
```diff
@@ -22,26 +23,37 @@
...
         try {
-            return new ResponseEntity<>(pdfService.convert(pdfRequest),
-                    pdfContentTypeHeader(), HttpStatus.OK);
+            return new ResponseEntity<>(pdfService.html2pdf(html), pdfContentTypeHeader(), OK);
         } catch (DavidPDFException e) {
```
[the remark this answers is not in the document]
Comment (KirkDerAuslaender) [reply src/main/java/de/ace/html2pdf/controller/PdfGenerationController.java:38]: Yes, it is a good idea. So we dont need to have a try catch.
Take a look @b1ank04 
We can just have on the DavidPDF thing a controlleradvice and then, there is no need of try catch anymore

File: src/main/java/de/ace/html2pdf/application/PdfService.java [hunk 6]
```diff
@@ -1,29 +1,91 @@
...
 @Slf4j
 @RequiredArgsConstructor
 public class PdfService {
 
     private final PdfRenderComponent pdfRenderComponent;
-    private final PostProcessorPdfComponent postProcessorPdfComponent;
 
-    public byte[] convert(PdfRequest pdfRequest) {
-        try (ByteArrayOutputStream bs = new ByteArrayOutputStream()) {
+    public byte[] html2pdf(String html) throws IOException {
```
Comment (DeepthiDNayak) [inline src/main/java/de/ace/html2pdf/application/PdfService.java:33]: This method throws IOException, but is not caught in the controller. We can either
 i) catch this exception in the service layer itself  OR 
 ii) propagate all the exceptions to Controller layer and catch it in the controller layer OR
  iii) propagate it to the GlobalExceptionHandler and let it handle the exception (we need an additional handler method)
  
  I am not sure which is the best approach here, may be @KirkDerAuslaender can comment more on it.

File: src/main/java/de/ace/html2pdf/config/exception/ExceptionHandler.java
Comment (DeepthiDNayak) [inline src/main/java/de/ace/html2pdf/config/exception/ExceptionHandler.java:1]: May be rename it to GlobalExceptionHandler, then we can use the @ExceptionHandler annotation without the full package name.

File: src/main/java/de/ace/html2pdf/config/exception/ExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.web.bind.annotation.ControllerAdvice;
+
+import java.nio.charset.StandardCharsets;
+
+import static org.springframework.http.HttpStatus.I_AM_A_TEAPOT;
+
+@ControllerAdvice
+public class ExceptionHandler {
+
+    @org.springframework.web.bind.annotation.ExceptionHandler(PdfException.class)
```
Comment (DeepthiDNayak) [inline src/main/java/de/ace/html2pdf/config/exception/ExceptionHandler.java:13]:
```suggestion
    @ExceptionHandler(PdfException.class)
```

File: src/main/java/de/ace/html2pdf/config/exception/PdfException.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package de.ace.html2pdf.config.exception;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Slf4j
+@ResponseStatus(value = HttpStatus.I_AM_A_TEAPOT)
```
Comment (DeepthiDNayak) [inline src/main/java/de/ace/html2pdf/config/exception/PdfException.java:8]: This is redundant, as we pass the status code from the exception handler method.

**Pull request merged**